src/lib/crypto/aes/operations.ts

```diff
@@ -0,0 +1,82 @@
+import { webcrypto } from 'one-webcrypto'
+import keys from './keys'
+import utils from '../utils'
+import { DEFAULT_SYMM_ALG, DEFAULT_CTR_LEN } from '../constants'
+import { SymmKey, SymmKeyOpts, SymmAlg, CipherText, Msg } from '../types'
+
+export async function encryptBytes(
+  msg: Msg,
+  key: SymmKey | string,
+  opts?: Partial<SymmKeyOpts>
+): Promise<CipherText> {
+  const data = utils.normalizeUtf16ToBuf(msg)
+  const importedKey = typeof key === 'string' ? await keys.importKey(key, opts) : key
+  const alg = opts?.alg || DEFAULT_SYMM_ALG
+  const iv = opts?.iv || utils.randomBuf(16)
+  const cipherBuf = await webcrypto.subtle.encrypt(
+    {
+      name: alg,
+      // AES-CTR uses a counter, AES-GCM/AES-CBC use an initialization vector
+      iv: alg === SymmAlg.AES_CTR ? undefined : iv,
+      counter: alg === SymmAlg.AES_CTR ? new Uint8Array(iv) : undefined,
+      length: alg === SymmAlg.AES_CTR ? DEFAULT_CTR_LEN : undefined,
+    },
+    importedKey,
+    data
+  )
+  return utils.joinBufs(iv, cipherBuf)
```

Comment:
We probably want to use a delimiter here like a period rather than just crushing the two together. It matches common behavior a bit more and provides a cleaner way to separate the two pieces of data later on.

Reply:
can do

Reply:
done